SU2_CFD/src/numerics/NEMO/CNEMONumerics.cpp

```diff
@@ -330,300 +334,54 @@ void CNEMONumerics::GetViscousProjFlux(su2double *val_primvar,
   }
 }
 
-void CNEMONumerics::GetViscousProjJacs(su2double *val_Mean_PrimVar,
-                                       su2double **val_Mean_GradPrimVar,
-                                       su2double *val_Mean_Eve,
-                                       su2double *val_Mean_Cvve,
-                                       su2double *val_diffusion_coeff,
-                                       su2double val_laminar_viscosity,
-                                       su2double val_eddy_viscosity,
-                                       su2double val_thermal_conductivity,
+#define COMPUTE_VISCOUS_JACS(NVAR, NSPECIES) \
+ComputeViscousJacs_impl<NVAR, NSPECIES>(val_Mean_PrimVar, \
+                                        val_Mean_Eve, val_Mean_Cvve, val_diffusion_coeff, \
+                                        val_laminar_viscosity, val_eddy_viscosity, \
+                                        val_thermal_conductivity, val_thermal_conductivity_ve, \
+                                        val_dist_ij, val_normal, val_dS, val_Fv, \
+                                        val_Jac_i, val_Jac_j, config);
+
+
+void CNEMONumerics::GetViscousProjJacs(const su2double *val_Mean_PrimVar,
+                                       su2double *val_Mean_Eve, const su2double *val_Mean_Cvve,
+                                       const su2double *val_diffusion_coeff, su2double val_laminar_viscosity,
+                                       su2double val_eddy_viscosity, su2double val_thermal_conductivity,
                                        su2double val_thermal_conductivity_ve,
-                                       su2double val_dist_ij, su2double *val_normal,
-                                       su2double val_dS, su2double *val_Fv,
+                                       su2double val_dist_ij, 
+                                       const su2double *val_normal,
+                                       su2double val_dS, const su2double *val_Fv,
                                        su2double **val_Jac_i, su2double **val_Jac_j,
-                                       const CConfig *config) {
+                                       const CConfig *config){
+
+  switch (nVar) {
+    case 5:
+      return COMPUTE_VISCOUS_JACS(5, 1);
 
+    case 6:
+      switch (nSpecies) {
+        case 1: return COMPUTE_VISCOUS_JACS(6, 1);
```

Comment:
Nice work :+1: this will probably not make an enormous difference, since the bottleneck is probably somewhere else, in any case, at least now there is an example in the code of how to do this.
It may feel cumbersome to do the manual switching over combinations of nVar/nSpecies, but the idea with this type of strategy is to template everything in the numerics, and only do the switching once at a higher level.